core/src/main/java/io/cloudevents/core/CloudEventUtils.java

```diff
@@ -102,4 +102,16 @@ public static CloudEvent toEvent(CloudEventReader reader, @Nullable CloudEventDa
         return mapper.map(data);
     }
 
+    /**
+     * Validates the extension name as defined in  CloudEvents spec
+     * @param name the extension name
+     * @return true if extension name is valid, false otherwise
+     */
+    public static boolean isValidExtensionName(String name) {
```

Comment:
Don't place it in utils, just make this a private method in `BaseCloudEventBuilder`